### PR TITLE
models added

### PR DESCRIPTION
**Pull request by RayaanRacer**

### CODE REVIEW

**Comment by RayaanRacer** [comment]
modelling done

File: models/User.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+// Schema
+
+const userSchema = new mongoose.Schema(
+    {   
+        u_id:{
+            type:String,
+            required:[true, "User Id is must"],
+            trim:true,
+        },
```
Comment (pranavz02) [inline models/User.js:11]: no need beacuse we have _id

File: models/User.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            default:false
+        },
+        isPassenger:{
+            type:Boolean,
+            default:false
+        },
+        isDriver:{
+            type:Boolean,
+            default:false
+        },
```
Comment (pranavz02) [inline models/User.js:75]: no need of all the criterias

File: models/User.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        createdAt:{
+            type:Date,
+            default:Date.now(),
+        }
+    }
```
Comment (pranavz02) [inline models/User.js:80]: instead use this
`
}, timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
`

File: models/Tarrif.js
Comment (pranavz02) [inline models/Tarrif.js:1]: write full names of the column or documents
eg carCategory, pricePerKM, pricePerExtraKM, etc.
removef required: true

File: models/Payment.js
Comment (pranavz02) [inline models/Payment.js:1]: write full names of the column or documents
eg paymentId, userID, driverID, etc
remove required: true

File: models/Payment.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    u_id:{
+        type:String,
+        required:[true,"Name/Id of associated user"],
+        trim:true,
+    },
+    d_id:{
+        type:String,
+        required:[true,"Name/Id of associated driver"],
+        trim:true,
+    },
```
Comment (pranavz02) [inline models/Payment.js:19]: connect the userID with the id of user in the UserModel. Do this same for driverID also
` userID: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'User',
        trim: true,
    }`

File: models/Payment.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    p_amt:{
+        type:Number,
+        required:[true,"Amount of payment"],
+        trim:true,
+    },
```
Comment (pranavz02) [inline models/Payment.js:24]: add document to connect with tarrifID 
`tariffID: {
type: mongoose.Schema.Types.ObjectId, 
required: true,
 ref: 'Tariff',
}`

File: models/Driver.js
Comment (pranavz02) [inline models/Driver.js:1]: write full names of the column or documents use camelCasing
eg DriverID, phone, licenceID, aadharID, 
no need for required true everywhere only important one - name, email, phone

add a column of password

File: models/Car.js
Comment (pranavz02) [inline models/Car.js:1]: write full names of the column or documents use camelCasing
carID, carModel, carCategory
remove required true from everywhere. Keep it only on necessary ones email, phone, name.

File: models/Car.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            type:String,
+            required:true,
+            trim:true,
+        },
+        driver_name:{
+            type:String,
+            required:true,
+            trim:true,
+        },
+    },
```
Comment (pranavz02) [inline models/Car.js:41]: instead connect it with driverModel
`driverID: { type: mongoose.Schema.Types.ObjectId,  required: true, ref: 'Driver', }`

File: models/Booking.js
Comment (pranavz02) [inline models/Booking.js:1]: write full names of the column or documents use camelCasing
eg bookingID, userID, etc
 remove required true from everywhere.

add document of paymentID as well and connect it with payment model

File: models/Booking.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+const BookingSchema = new mongoose.Schema({
+    b_id:{
+        type:String,
+        required:true,
+    },
+    u_id:{
+        type:String,
+        required:true,
+    },
```
Comment (pranavz02) [inline models/Booking.js:13]: connect it with userModel
` userID: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'User',
    }`

File: models/Booking.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    carCat:{
+        type:String,
+        required:[true, "Which car to choose"],
+        trim:true,
+        enum:{
+            values: ["sedan","mini suv 4 seater","mini suv 7 seater","suv","premium sedan","premium suv","tampo traveller 13 seater","tampo traveller 17 seater"],
+            message:`Car is not supported`,
+        },
+    },
```
Comment (pranavz02) [inline models/Booking.js:32]: connect it with car model

File: models/Booking.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    nightAllowance:{
+        type:Number,
+        required:[true,"provide Night allowance data for driver if applicable"],
+        trim:true,
+    },
+    gst:{
+        type:Number,
+        required:[true,"provide total expected gst"],
+        trim:true,
+    },
```
Comment (pranavz02) [inline models/Booking.js:75]: connect it with tariffID
we will get all information from there.

File: models/Booking.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    expectedCompleteCharge:{
```
Comment (pranavz02) [inline models/Booking.js:76]: make it approxAmount

File: models/Booking.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        type:Number,
+        required:[true,"Expected Complete Charge"],
+        trim:true,
+    },
+    actualCompleteCharge:{
```
Comment (pranavz02) [inline models/Booking.js:81]: make it finalAmount

**Review by pranavz02 — CHANGES_REQUESTED** [review]
change the created_at document in every model to this
` }, timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }`

**Pull request merged**